base/shell/explorer/shellservice.cpp

```diff
@@ -112,6 +114,7 @@ HRESULT InitShellServices(HDPA * phdpa)
     {
         ERR("RegEnumValueW failed %08x.\n", ret);
         hr = HRESULT_FROM_WIN32(GetLastError());
+        RegCloseKey(hkey);
```

Comment:
Just do it on cleanup if `hkey` is not NULL?

Reply:
You're right and no need to check since both previous RegCloseKey and failed RegOpenKeyW lead to a return condition preventing entering Goto cleanup.

Change done.